azurelinuxagent/common/protocol/imds.py

```diff
@@ -254,38 +261,69 @@ def _http_get(self, endpoint, resource_path, headers):
         url = self._get_metadata_url(endpoint, resource_path)
         return restutil.http_get(url, headers=headers, use_proxy=False)
 
+    def _get_metadata_from_endpoint(self, endpoint, resource_path, headers):
+        """
+        Get metadata from one of the IMDS endpoints.
+
+        :param str endpoint: IMDS endpoint to call
+        :param str resource_path: path of IMDS resource
+        :param bool headers: headers to send in the request
+        :return: Tuple<status:int, response:str>
+            status: one of the following response status codes: IMDS_RESPONSE_SUCCESS, IMDS_RESPONSE_ERROR,
+                    IMDS_CONNECTION_ERROR, IMDS_INTERNAL_SERVER_ERROR
+            response: IMDS response on IMDS_RESPONSE_SUCCESS, failure message otherwise
+        """
+        try:
+            resp = self._http_get(endpoint=endpoint, resource_path=resource_path, headers=headers)
+        except ResourceGoneError:
+            return IMDS_INTERNAL_SERVER_ERROR, "IMDS error in /metadata/{0}: HTTP Failed with Status Code 410: Gone".format(resource_path)
+        except HttpError as e:
+            msg = str(e)
```

Comment:
`ustr` instead of `str`?

Reply:
I'm not sure if that change would break the regex matches that follow.

Reply:
Unless you are not logging the exact message, `str` should be good rather than `ustr`. Just do a quick check to see is the message is being translated into `ustr` before logging.